### ISSUE TITLE
Implement DIMER BYOD Prithvi-EO-2.0 pixelwise regression pipeline

### ISSUE THREAD

**Issue by kurtvalcorza** [kurtvalcorza/prithvi-eo-regression-pipeline#1]
## Goal
Build the umbrella/integration repository for a DIMER BYOD **Earth-observation pixelwise regression** workflow using Prithvi-EO-2.0 + TerraTorch.

User flow:

**Upload multispectral EO dataset → validate geospatial/band/target alignment → configure → fine-tune → evaluate → export model**.

Split deployable repos:
- `kurtvalcorza/prithvi-eo-regression-dataset-validator`
- `kurtvalcorza/prithvi-eo-regression-finetuner`

## Upstream baseline
Prithvi-EO-2.0 is an IBM/NASA multi-temporal EO foundation-model family trained on HLS imagery. Official downstream examples use TerraTorch. The upstream `biomassters.yaml` uses `PixelwiseRegressionTask`, six HLS/Sentinel-2-like bands, temporal inputs, an encoder-decoder architecture, AdamW and mixed precision.

Initial DIMER base model for v1:
- `ibm-nasa-geospatial/Prithvi-EO-2.0-100M-TL`
- Apache-2.0

Use the 100M-TL variant first to keep DIMER GPU requirements tractable. Larger 300M/600M variants are follow-ups only after resource profiling.

References:
- https://github.com/NASA-IMPACT/Prithvi-EO-2.0
- https://github.com/NASA-IMPACT/Prithvi-EO-2.0/blob/main/configs/biomassters.yaml
- https://huggingface.co/ibm-nasa-geospatial/Prithvi-EO-2.0-100M-TL

## DIMER Pipeline Builder metadata
Document:
- **Pipeline Name:** `Prithvi-EO Pixelwise Regression`
- **Description:** `Fine-tune Prithvi-EO-2.0 for pixelwise regression on multispectral Earth-observation imagery using your own geospatial dataset. Supports HLS-style spectral bands, optional temporal stacks, dataset validation, configurable fine-tuning, evaluation, and export of the trained model.`
- **Task Type:** `Custom / Other`
- **Base Model:** `ibm-nasa-geospatial/Prithvi-EO-2.0-100M-TL`
- **Validator Repository:** `https://github.com/kurtvalcorza/prithvi-eo-regression-dataset-validator`
- **Finetuner Repository:** `https://github.com/kurtvalcorza/prithvi-eo-regression-finetuner`

Use a baked fallback such as `DIMER_TASK_TYPE=eo_pixelwise_regression` because Custom/Other currently resolves generically.

## Task definition
V1 is **pixelwise/dense regression**, not scalar scene-level regression. Each input chip/temporal stack maps to a spatially aligned continuous-valued target raster. Above-ground biomass estimation (BioMassters) is the canonical upstream-style example.

Do not mix scalar metadata regression and dense raster regression in v1; they require different heads/dataset contracts.

## Canonical dataset contract
Define an explicit manifest/index-driven format rather than relying on folder guessing. Recommended package:

```text
manifest.csv
images/
  <sample>_t0.tif
  <sample>_t1.tif ...
targets/
  <sample>_target.tif
```

Manifest fields should identify:
- sample id
- split (optional train/val/test)
- input raster path(s) and temporal order/date
- target raster path
- band mapping/order
- optional geolocation/temporal metadata

Supported raster format: GeoTIFF first. Preserve CRS/transform/nodata metadata.

Initial canonical spectral order should align with Prithvi/HLS expectations:
- BLUE
- GREEN
- RED
- NIR_NARROW
- SWIR_1
- SWIR_2

Allow explicit band mapping rather than assuming band positions blindly.

## Validation requirements
Cross-repo contract must cover:
- readable GeoTIFFs
- expected band count/order
- image-target width/height alignment
- CRS/transform/pixel-grid alignment
- finite/nodata target handling
- input/target sample pairing
- temporal ordering and frame-count consistency
- train/val/test sample disjointness
- no accidental duplicate chips across splits
- dtype/range diagnostics
- geospatial metadata availability
- resource/archive security limits

## Preprocessing controls
Candidate manifest fields:
- `bands`
- `num_frames`
- `image_size`
- `nodata_strategy`
- `rescale`
- `use_temporal_location_embeddings`
- `validation_fraction`
- `test_fraction`
- augmentation toggles that are genuinely implemented

Do not silently alter spectral order or fabricate missing bands.

## Fine-tuning controls
Candidate fields based on TerraTorch/official configs:
- `epochs`
- `batch_size`
- `learning_rate`
- `weight_decay`
- `optimizer`
- `precision`
- `freeze_backbone`
- `freeze_decoder`
- `decoder` (keep allowlisted/tested; default UperNet-style path if supported for selected model)
- scheduler parameters where implemented
- `seed`

DIMER Base Model is authoritative; no `model_id` in `dimer-pipeline.json`.

## Evaluation
For valid target pixels report:
- RMSE
- MAE
- R² where statistically meaningful
- bias / mean error
- valid-pixel count
- per-scene and aggregate metrics

Respect nodata masks. Do not score invalid/masked pixels as zeros.

## Artifact contract
Export a reloadable model bundle containing:
- trained checkpoint
- Prithvi base model id + revision
- TerraTorch/PyTorch versions
- decoder/task configuration
- band mapping
- temporal/frame configuration
- geospatial preprocessing/nodata policy
- normalization/rescaling metadata
- hyperparameters and seed
- metrics
- provenance/checksums
- inference example producing a georeferenced prediction raster where feasible

Smoke path: **fine-tune → save → reload → predict raster**.

## Canonical open data
Primary upstream-style reference: **BioMassters** above-ground biomass estimation (CC BY 4.0), for which the Prithvi repo provides an official TerraTorch pixelwise regression configuration. The full dataset is ~201 GB, so DIMER smoke testing must use a deterministic, attributed small subset only.

CI should rely on tiny synthetic GeoTIFF fixtures generated in tests.

## Acceptance criteria
- [ ] Pixelwise regression task boundary documented.
- [ ] DIMER metadata documented.
- [ ] GeoTIFF/manifest BYOD format documented.
- [ ] Spectral band and temporal semantics explicit.
- [ ] Geo-alignment/nodata rules explicit.
- [ ] Root `validate.py` / root `train.py` contracts enforced in split repos.
- [ ] Finetuner root `dimer-pipeline.json`; runtime parity enforced.
- [ ] No manifest `model_id`.
- [ ] Safe Custom/Other task fallback.
- [ ] Official TerraTorch Prithvi path used rather than bespoke backbone code unless strictly necessary.
- [ ] Dense regression metrics respect nodata masks.
- [ ] Reloadable checkpoint/predict smoke passes.
- [ ] Docker builds + CI pass.
- [ ] DIMER validation/training smoke passes.
- [ ] Enable checklist reaches 10/10 before enablement.

## Non-goals v1
- scene-level scalar regression
- arbitrary sensors/band sets without explicit mapping/support
- 300M/600M variants before GPU profiling
- distributed training/HPO.
